### ISSUE TITLE
[cssom] Should shorthands in object from getComputedStyle be enumerable?

### ISSUE THREAD

**Issue by upsuper** [w3c/csswg-drafts#2877]
In #2529 we resolved that object returned from gCS should include shorthands, but it is not clear whether they should be enumerable from from that object.

I think I saw someone iterating the properties from gCS result to copy style of one element to another. If we have every shorthands listed there, there is a risk that stuff would get overridden if the order happens to be unfortunate (shorthand comes after its longhands?)

It was raised during the discussion of #2529.

**Comment by tabatkins** [comment]
Chrome currently does *not* include shorthands in the list of properties when you enumerate them:

[Example](http://software.hixie.ch/utilities/js/live-dom-viewer/saved/6024)
```
<!DOCTYPE html>
<body>
<script>
var x = [...getComputedStyle(document.body)];
w(x);
</script>
```

**Comment by gsnedders** [comment]
We should probably add some idlharness tests into each spec directory for the properties they define.

**Comment by gsnedders** [comment]
What type of enumerability are we talking about here?

Are we talking about `[...getComputedStyle(document.body)]` per @tabatkins's comment, or are we talking about `Object.getOwnPropertyDescriptor(CSSStyleDeclaration.prototype, "font").enumerable)`?

**Comment by Loirooriol** [comment]
It seems clear to me that `CSSStyleDeclaration` only contains longhands, e.g. [`setProperty`](https://drafts.csswg.org/cssom/#dom-cssstyledeclaration-setproperty) takes care of expanding shorthands. Then, `length` and `item()` should only account for longhands.

`item()` is an [indexed property getter](https://webidl.spec.whatwg.org/#dfn-indexed-property-getter), so per https://webidl.spec.whatwg.org/#es-iterable, `@@iterator` should just be `%Array.prototype.values%`.

So it follows that `[...getComputedStyle(document.body)]` should not include shorthands either.

Regarding enumeration:

```js
var arr = [];
for (let p in getComputedStyle(document.body)) arr.push(p);
arr;
```

https://webidl.spec.whatwg.org/#define-the-attributes says `[[Enumerable]]: true`, so regardless of whether they are longhands or shorthands, the above should include:
 - [camel-cased attributes](https://drafts.csswg.org/cssom/#dom-cssstyledeclaration-camel-cased-attribute) like `WebkitBorderRadius`
 - [webkit-cased attributes](https://drafts.csswg.org/cssom/#dom-cssstyledeclaration-webkit-cased-attribute) like `webkitBorderRadius`
 - [dashed attributes](https://drafts.csswg.org/cssom/#dom-cssstyledeclaration-dashed-attribute) like `-webkit-border-radius`

That's the case in Gecko and WebKit, but Blink only enumerates `webkitBorderRadius`.

**Comment by Loirooriol** [comment]
And BTW, this should be the same for both `document.body.style` and `getComputedStyle(document.body)`.

**Comment by shivamidow** [comment]
> Chrome currently does _not_ include shorthands in the list of properties when you enumerate them:

No, Chrome includes `font-variant` in the enumerated properties, which is a shorthand. I see `font-variant` in your outputs either.
